src/org/zaproxy/zap/extension/alert/ExtensionAlert.java

```diff
@@ -284,6 +284,14 @@ private void publishAlertEvent(Alert alert, String event) {
         Map<String, String> map = new HashMap<>();
         map.put(AlertEventPublisher.ALERT_ID, Integer.toString(alert.getAlertId()));
         map.put(AlertEventPublisher.HISTORY_REFERENCE_ID, Integer.toString(alert.getSourceHistoryId()));
+        map.put(AlertEventPublisher.NAME, alert.getName());
+        map.put(AlertEventPublisher.URI, alert.getUri().toString());
+        map.put(AlertEventPublisher.PARAM, alert.getParam());
+        map.put(AlertEventPublisher.RISK, Integer.toString(alert.getRisk()));
+        map.put(AlertEventPublisher.RISK_STRING, Alert.MSG_RISK[alert.getRisk()]);
+        map.put(AlertEventPublisher.CONFIDENCE, Integer.toString(alert.getConfidence()));
+        map.put(AlertEventPublisher.CONFIDENCE_STRING, Alert.MSG_CONFIDENCE[alert.getConfidence()]);
+        map.put(AlertEventPublisher.SOURCE, alert.getSource().getI18nKey());

```

Comment:
I'd suggest returning the ID instead, the key is more likely to change than the ID.